components/browser/icons/src/main/java/mozilla/components/browser/icons/processor/DiskIconProcessor.kt

```diff
@@ -27,10 +35,12 @@ class DiskIconProcessor(
         icon: Icon,
         desiredSize: DesiredSize
     ): Icon {
-        if (resource != null && icon.shouldCacheOnDisk && !request.isPrivate) {
-            cache.put(context, request, resource, icon)
+        if (resource != null && !request.isPrivate) {
+            cache.putResources(context, request)
```

Comment:
Overall this looks good. I wonder if we can find a way to optimize this `putResources` away in situations where we already have them on disk? Otherwise we will always perform a disk write here which seems wasteful?

However I am not sure if there is a good way to do that. This would require that we know here (or inside the cache) whether there are new resources or if they all have been loaded from memory (like the icon). Do you think it would work if we'd [change the order of the processors here](https://github.com/mozilla-mobile/android-components/blob/master/components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt#L96-L99) to first execute the disk processor and then the memory processor? Then we could maybe check here if the mapping is already in the memory cache? And then we know we also wrote it to disk already?

Or maybe you have another idea?

Anyhow, if that is not possible or too complicated then I guess we could also land as-is.

Reply:
But the resources are just 
  ```
data class Resource(
        val url: String,
        val type: Type,
        val sizes: List<HtmlSize> = emptyList(),
        val mimeType: String? = null,
        val maskable: Boolean = false
    )
```

serialized to JSON, not the actual bitmaps, and they're saved to a `DiskLruCache` with maxsize of `MAXIMUM_CACHE_RESOURCES_BYTES: Long = 1024 * 1024 * 10 // 10 MB`

I'm not entirely sure if changing the processors order would help. But I'd need to investigate some more to be sure.